src/ad/azext_ad/tests/latest/preparers.py

```diff
@@ -0,0 +1,159 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+
+import os
+from datetime import datetime
+from azure_devtools.scenario_tests import SingleValueReplacer
+from azure.cli.testsdk.preparers import NoTrafficRecordingPreparer
+from azure.cli.testsdk.exceptions import CliTestError
+from azure.cli.testsdk.reverse_dependency import get_dummy_cli
+
+
+KEY_RESOURCE_GROUP = 'rg'
+KEY_VIRTUAL_NETWORK = 'vnet'
+KEY_VNET_SUBNET = 'subnet'
+KEY_VNET_NIC = 'nic'
+
+
+class VirtualNetworkPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
```

Comment:
Not used? I can't seem to find any usage of `VirtualNetworkPreparer`.

Reply:
Auto-generated by CodeGen. I didn't use it but keep it because it will always be generated. No need to delete every time.

Reply:
@changlong-liu could we remove these preparer and import from azure cli core module?

Reply:
yes @Juliehzl , we are refactoring the test preparers codegen.

Hi @evelyn-ys , In latest autorest.az the preparers.py is supposed to be generated only when has some reference for it. Please help to send me the swagger branch to have further check if it generated even have no usage.

Reply:
@changlong-liu Hope [this stable swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/domainservices/resource-manager/Microsoft.AAD/stable/2020-01-01/domainservices.json) can help